daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/ContractStateMachine.scala

```diff
@@ -121,38 +123,41 @@ object ContractStateMachine {
 
     /** Visit a create node */
     def handleCreate(node: Node.Create): Either[KeyInputError, State[Nid]] =
-      visitCreate(node.coid, node.gkeyOpt).left.map(Right(_))
+      visitCreate(node.coid, node.gkeyOpt)
 
     private[lf] def visitCreate(
         contractId: ContractId,
         mbKey: Option[GlobalKey],
-    ): Either[DuplicateContractKey, State[Nid]] = {
-      val me =
-        this.copy(
-          locallyCreated = locallyCreated + contractId,
-          activeState = this.activeState
-            .copy(locallyCreatedThisTimeline =
-              this.activeState.locallyCreatedThisTimeline + contractId
-            ),
-        )
-      // if we have a contract key being added, include it in the list of
-      // active keys
-      mbKey match {
-        case None => Right(me)
-        case Some(gk) =>
-          val conflict = lookupActiveKey(gk).exists(_ != KeyInactive)
-
-          val newKeyInputs =
-            if (globalKeyInputs.contains(gk)) globalKeyInputs
-            else globalKeyInputs.updated(gk, KeyCreate)
-          Either.cond(
-            !conflict || mode == ContractKeyUniquenessMode.Off,
-            me.copy(
-              activeState = me.activeState.createKey(gk, contractId),
-              globalKeyInputs = newKeyInputs,
-            ),
-            DuplicateContractKey(gk),
+    ): Either[CreateError, State[Nid]] = {
+      if (locallyCreated.contains(contractId)) {
```

Comment:
we should also check vs, contract already fetched !

Reply:
Done